spec/requests/incident_spec.rb

```diff
@@ -280,108 +274,106 @@ def breadcrumbs_classes
-      login(user: @bob)
+      login(user: bob)
       expect(status_nav_count(:draft)).to eq(0)
       expect(status_nav_count(:in_review)).to eq(0)
       # But he can see his own
-      create_partial_incident :review
-      create_complete_incident
+      create(:incident, user_id: bob.user_id, submit: false)
+      create(:incident, user_id: bob.user_id, submit: true)
+      visit dashboard_path
       expect(status_nav_count(:draft)).to eq(1)
       expect(status_nav_count(:in_review)).to eq(1)
       logout
 
       # Alice can't see Bob's new incidents
-      login(user: @alice)
+      login(user: alice)
       expect(status_nav_count(:draft)).to eq(1)
       expect(status_nav_count(:in_review)).to eq(1)
       logout
 
       # Claire (the admin) can see BOTH of their complete incidents,
       # but neither of their incomplete incidents.
-      login(user: @claire)
+      login(user: claire)
       expect(status_nav_count(:draft)).to eq(0)
       expect(status_nav_count(:in_review)).to eq(2)
       # Claire can see and review her own incidents
-      create_partial_incident :review
-      create_complete_incident
+      create(:incident, user_id: claire.user_id, submit: false)
+      create(:incident, user_id: claire.user_id, submit: true)
+      visit dashboard_path
       expect(status_nav_count(:draft)).to eq(1)
       expect(status_nav_count(:in_review)).to eq(3)
       logout
 
       # Alice can't see Claire's incidents
-      login(user: @alice)
+      login(user: alice)
       expect(status_nav_count(:draft)).to eq(1)
       expect(status_nav_count(:in_review)).to eq(1)
       logout
 
       # Bob can't see Claire's incidents
-      login(user: @bob)
+      login(user: bob)
       expect(status_nav_count(:draft)).to eq(1)
       expect(status_nav_count(:in_review)).to eq(1)
       logout
 
       # Dan can't see anyone's incidents
-      login(user: @dan)
+      login(user: dan)
       expect(status_nav_count(:draft)).to eq(0)
       expect(status_nav_count(:in_review)).to eq(0)
       logout
     end
 
     it 'lets an administrator edit another employee\'s incident' do
-      login(user: @alice)
-      create_complete_incident
+      login(user: alice)
+      create(:incident, user_id: alice.user_id)
       logout
 
-      login(user: @claire)
+      login(user: claire)
       visit_status :in_review
       find('a', text: 'View').click
       expect(current_path).to end_with('/review')
-      expect(page).to have_content('White')
-      expect(page).to have_no_content('Cambodian')
+      expect(page).to have_no_content('Female')
 
       find('h1', text: 'Officer').find('~a', text: 'Edit').click
       expect(current_path).to end_with('/involved_officers/0/edit')
-      check 'Cambodian'
+      choose 'Female'
       click_button 'Save and Continue'
-      expect(page).to have_content('White')
-      expect(page).to have_content('Cambodian')
+      expect(page).to have_content('Female')
```

Comment:
Race is nondeterministic when we use the `incident` factory, but gender is one of the deterministic fields.